### ISSUE TITLE
Boolean calculated property renderer/editor doesn't work as expected

### ISSUE THREAD

**Issue by karolis-zukauskas** [iTwin/presentation#732]
## Issue
The following ruleset creates a calculated **boolean** property that renders correctly ("True" / "False"), but the editor shows a string editor: 
``` ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
![Image](https://github.com/user-attachments/assets/3eb147d3-e104-4629-9f65-bde61312e93d)

The same rule, just with an explicit `type: "bool"` renders incorrectly, but has the correct editor for a boolean property (a checkbox):
``` ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      type: "bool",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
![Image](https://github.com/user-attachments/assets/f9b502e1-c000-4104-9b66-5b622f388c94)

## Expected behaviour
Calculated boolean property should render with "True" / "False" values and editor for such property should be a checkbox.

## Versions
"@itwin/presentation-backend": "4.9.1",
"@itwin/presentation-common": "4.9.1",
"@itwin/presentation-frontend": "4.9.1",
"@itwin/presentation-components": "5.4.1",

**Comment by JonasDov** [comment]
Regarding this ruleset:
```ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
It works as expected, since calculated property is by default created as a `string` value, see: https://www.itwinjs.org/presentation/content/calculatedpropertiesspecification/#attribute-type. If you want calculated property to have a specific type, then you should specify it by adding `type` attribute.

Regarding this ruleset:
```ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      type: "bool",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
This issue is fixed in https://github.com/iTwin/imodel-native/pull/889

**Comment by JonasDov** [comment]
> ## Issue
> The following ruleset creates a calculated **boolean** property that renders correctly ("True" / "False"), but the editor shows a string editor:
> 
> {
>   ruleType: "ContentModifier",
>   class: { schemaName: "BisCore", className: "Element" },
>   calculatedProperties: [
>     {
>       label: "Test Aspect Property",
>       value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
>     }
>   ],
> }
> ![Image](https://github.com/user-attachments/assets/3eb147d3-e104-4629-9f65-bde61312e93d)
> 
> The same rule, just with an explicit `type: "bool"` renders incorrectly, but has the correct editor for a boolean property (a checkbox):
> 
> {
>   ruleType: "ContentModifier",
>   class: { schemaName: "BisCore", className: "Element" },
>   calculatedProperties: [
>     {
>       label: "Test Aspect Property",
>       type: "bool",
>       value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
>     }
>   ],
> }
> ![Image](https://github.com/user-attachments/assets/f9b502e1-c000-4104-9b66-5b622f388c94)
> 
> ## Expected behaviour
> Calculated boolean property should render with "True" / "False" values and editor for such property should be a checkbox.
> 
> ## Versions
> "@itwin/presentation-backend": "4.9.1", "@itwin/presentation-common": "4.9.1", "@itwin/presentation-frontend": "4.9.1", "@itwin/presentation-components": "5.4.1",

Regarding this ruleset:
```ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
It works as expected, since calculated property is by default created as a `string` value, see: https://www.itwinjs.org/presentation/content/calculatedpropertiesspecification/#attribute-type. If you want calculated property to have a specific type, then you should specify it by adding `type` attribute.

Regarding this ruleset:
```ts
{
  ruleType: "ContentModifier",
  class: { schemaName: "BisCore", className: "Element" },
  calculatedProperties: [
    {
      label: "Test Aspect Property",
      type: "bool",
      value: "this.HasRelatedInstance(\"BisCore:ElementOwnsUniqueAspect\", \"Forward\", \"BisCore:ElementUniqueAspect\")",
    }
  ],
}
```
This issue is fixed in https://github.com/iTwin/imodel-native/pull/889

**Comment by grigasp** [comment]
@karolis-zukauskas, the fix has been released with `itwinjs-core` version `4.9.5`, it's on the backend (native addon).
